### PR TITLE
drivers: MXSMIO SPI flash support

### PR DESCRIPTION
**Pull request by temonuv**
SPI flash sub-driver to support MXSMIO SPI flash.
It is based on the WinBond SPI flash driver.

Signed-off-by: Piotr Błachnio <piotr.blachnio@sigmaconnectivity.se>

### CODE REVIEW

**Comment by temonuv** [comment]
Dear reviewers can you give me advice what is wrong with "Identity/Emails — Verification failed" ?

**Comment by carlescufi** [comment]
@temonuv make sure that your Git identity matches the one in the `Signed-off-by` lines. To verify your Git identity, run:

```
git config user.name
git config user.email
```
These need to match the sign-off.

**Comment by temonuv** [comment]
@carlescufi 
Looks like it shouldn't be a problem
$ git config user.name
Piotr Błachnio
$ git config user.email 
piotr.blachnio@sigmaconnectivity.se

Can it be that I have different e-mail on github than in sign-off and commit msg. ?

**Comment by carlescufi** [comment]
@temonuv I think the issue is the following:

```
Author: piotr.blachnio <piotr.blachnio@sigmaconnectivity.se>
Date:   Tue Nov 27 09:48:22 2018 +0100

    samples: drivers/spi_flash_for_pca10056

    Add sample application for MX25R6435F - external flash memory placed
    on Nordic Semiconductors eval board PCA10056

    Boards dts was changes to make sample working without further changes
    and still pins <38> and <39> can be use as I2C.

    Signed-off-by: Piotr Błachnio <piotr.blachnio@sigmaconnectivity.se>
```
You can see there that the `Author` line does not match the `Signed-off-by` line:

`Author: piotr.blachnio <piotr.blachnio@sigmaconnectivity.se>`
vs
`Signed-off-by: Piotr Błachnio <piotr.blachnio@sigmaconnectivity.se>`

You can use `git commit --amend --author` to fix this

File: boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts [hunk 1]
```diff
@@ -126,9 +126,9 @@
 
 &spi2 {
 	status = "ok";
-	sck-pin = <37>;
```
Comment (nvlsianpu) [inline boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts]: Cześć! This change should be done in DTS overlay file belongs to the sample project. no need to change board DTS which might accidental impose badly other samples.

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts (hunk 1 above)
Comment (temonuv) [reply boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts]: I have added overlay.

**Comment by temonuv** [comment]
@vanwinkeljan no :( I didn't notice existence of this driver.

**Pull request closed without merging**